go/vt/vtgate/engine/gen4_compare_v3.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"encoding/json"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/log"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+// Gen4CompareV3 is a Primitive used to compare V3 and Gen4's plans.
+type Gen4CompareV3 struct {
+	V3, Gen4   Primitive
+	HasOrderBy bool
+}
+
+var _ Primitive = (*Gen4CompareV3)(nil)
+var _ Gen4Comparer = (*Gen4CompareV3)(nil)
+
+// GetGen4Primitive implements the Gen4Comparer interface
```

Comment:
nit: since the field is public, and this is not an interface, might as well access the field directly, right?